src/app/nonprofit/nonprofit-page/donation-dialog.html

```diff
@@ -0,0 +1,124 @@
+<button mat-icon-button class="close-icon-button" [mat-dialog-close]="true">
+    <mat-icon class="close-icon">close</mat-icon>
+</button>
+<h1 mat-dialog-title>
+    Thank you for choosing to to donate to {{data.nonprofitName}}!
+</h1>
+<div mat-dialog-content class="donation-stepper-container">
+    <mat-vertical-stepper linear #stepper>
+        
+        <!-- Donation Amount Stepper-->
+        <mat-step [stepControl]="donationAmountForm">
+            <form [formGroup]="donationAmountForm">
+                <ng-template matStepLabel>Enter Donation Amount</ng-template>
+
+                <div class="donation-inputs-container">
+                    <!-- Input for donation amount -->
+                    <div class="donation-input">
+                        <mat-form-field>
+                            <mat-label>Enter Amount to Donate</mat-label>
+                            <input formControlName="donationAmount" matInput type="tel" currencyMask>
```

Comment:
I noticed this too. This happens because the currency mask library that I'm using initially sets some variables to null/nothing. However, after the user inputs a number and then deletes it, it sets the variable to 0. So even though technically the user has not entered a "valid" currency, it will accept it because 0 is an acceptable value. So I did some research and there are 3 ways to fix this, I can spend a few days creating a custom error handler for it, I can get rid of the currency mask for this input (easiest one), OR we can just leave it as is. I think option 2 is the best one but what do you think?

Reply:
Is it possible to just do something with not allowing the form control value being equal to zero just to handle that specific situation? If that would take too much time, then it should be fine to leave it as is and we can go back and fix it if we have more time later, just add it as an issue in the repo or mark it with a TODO comment

Reply:
I tried to fix it and although it is possible, I think it would be too much work for now. Maybe we can come back to it later but I couldn't find a quick fix for this. This was actually one of the cons people mentioned about working with stepper-forms in Angular. it's apparently not easy to deal with validation between steppers